File: TransCoder_tokenized_test_set_functions/cpp/MAXIMIZE_ARRJ_ARRI_ARRL_ARRK_SUCH_THAT_I_J_K_L.cpp
- 100000000 int findMaxValue ( int arr [ ], int n ) {
  if ( n < 4 ) {
    printf ( "The array should have atlest 4 elements\n" );
    return MIN;
  }
  int table1 [ n + 1 ], table2 [ n ], table3 [ n - 1 ], table4 [ n - 2 ];
  for ( int i = 0;
  i <= n;
  i ++ ) table1 [ i ] = table2 [ i ] = table3 [ i ] = table4 [ i ] = MIN;
  for ( int i = n - 1;
  i >= 0;
  i -- ) table1 [ i ] = max ( table1 [ i + 1 ], arr [ i ] );
  for ( int i = n - 2;
  i >= 0;
  i -- ) table2 [ i ] = max ( table2 [ i + 1 ], table1 [ i + 1 ] - arr [ i ] );
  for ( int i = n - 3;
  i >= 0;
  i -- ) table3 [ i ] = max ( table3 [ i + 1 ], table2 [ i + 1 ] + arr [ i ] );
  for ( int i = n - 4;
  i >= 0;
  i -- ) table4 [ i ] = max ( table4 [ i + 1 ], table3 [ i + 1 ] - arr [ i ] );
  return table4 [ 0 ];
}

File: TransCoder_tokenized_test_set_functions/python/MAXIMIZE_ARRJ_ARRI_ARRL_ARRK_SUCH_THAT_I_J_K_L.py
def findMaxValue ( arr, n ) :
    if n < 4 :
        print ( "The array should have atlest 4 elements" )
        return MIN
    table1, table2 = [ MIN ] * ( n + 1 ), [ MIN ] * n
    table3, table4 = [ MIN ] * ( n - 1 ), [ MIN ] * ( n - 2 )
    for i in range ( n - 1, - 1, - 1 ) :
        table1 [ i ] = max ( table1 [ i + 1 ], arr [ i ] )
    for i in range ( n - 2, - 1, - 1 ) :
        table2 [ i ] = max ( table2 [ i + 1 ], table1 [ i + 1 ] - arr [ i ] )
    for i in range ( n - 3, - 1, - 1 ) :
        table3 [ i ] = max ( table3 [ i + 1 ], table2 [ i + 1 ] + arr [ i ] )
    for i in range ( n - 4, - 1, - 1 ) :
        table4 [ i ] = max ( table4 [ i + 1 ], table3 [ i + 1 ] - arr [ i ] )
    return table4 [ 0 ]
